### PR TITLE
Use mmap to load the weights

### PR DESCRIPTION
**Pull request by howard0su**

### CODE REVIEW

File: include/rwkv/cuda/rwkv.cu [hunk 1]
```diff
@@ -619,84 +620,73 @@ unsigned long long getSize(unsigned long long i, unsigned long long n_layers, un
...
     
   
+    char* current = (char*)map.addr + sizeof(int64_t) * 2;
+
+    for(int64_t i = 0; i < 46; i++) {
+        char* buffer;
+        int64_t size = getSize(i, n_layers, n_embed);
+
+        if (i == EMBED)
+            continue; // keep EMB on CPU
```
Comment (harrisonvanderbyl) [inline include/rwkv/cuda/rwkv.cu:659]: will you need to set ptrs[EMBED] = something here?

Comment (harrisonvanderbyl) [reply include/rwkv/cuda/rwkv.cu:659]: Also, EMBED is undefined in this context. Perhaps the enum needs to be moved to a separate enums.h file?

Comment (howard0su) [reply include/rwkv/cuda/rwkv.cu:659]: You are right. I will fix it.

**Comment by nenkoru** [comment]
Result is after a few re-runs:
1B5
this PR: Loading time 1.9288136959075928s
current main: Loading time 1.9180498123168945s

3B
this PR: Loading time 2.8054773807525635s
current main: Loading time 2.8033857345581055s

128GB of DDR4 RAM
Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz
fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm cpuid_fault epb invpcid_single pti intel_ppin ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust b mi1 avx2 smep bmi2 erms invpcid cqm xsaveopt cqm_llc cqm_occup_llc dtherm ida arat pln pts md_clear flush_l1d

**Pull request closed without merging**